src/packages/admin-ui-components/src/detail-panel/component.tsx

```diff
@@ -72,24 +72,57 @@ const SelectField = ({ name, entity }: { name: string; entity: EntityField }) =>
 	);
 };
 
+const BooleanField = ({ name }: { name: string }) => {
+	const [_, meta, helpers] = useField({ name, multiple: false });
+	const { initialValue } = meta;
+
+	useEffect(() => {
+		helpers.setValue(initialValue.value ? initialValue.value : undefined);
+	}, []);
+
+	const handleOnChange = (selected: SelectOption[]) => {
+		const value = selected?.[0]?.value;
+		if (value === undefined) {
+			helpers.setValue(undefined);
+		} else {
+			helpers.setValue(value);
+		}
+	};
+
+	return (
+		<Select
```

Comment:
Why have a select component do a boolean thing? Isn't a checkbox better for handling a boolean use case?

Reply:
yeah I agree it would be better. We would need one that can cater for three states true, false and undefined. Something that we can look at adding later.